pkg/resources/reconciling/ensure.go

```diff
@@ -60,6 +62,49 @@ func createWithName(rawcreate ObjectCreator, name string) ObjectCreator {
 	}
 }
 
+// EnsureObject is specification of the object to ensure.
+type EnsureObject struct {
+	Name             types.NamespacedName
+	Creator          ObjectCreator
+	EmptyObj         runtime.Object
+	RequiresRecreate bool
+}
+
+// EnsureNamedObjectsConcurrent calls `EnsureNamedObject` for each specifiecation concurrently and returns aggregated errors if any occurred.
+func EnsureNamedObjectsConcurrent(ctx context.Context, client ctrlruntimeclient.Client, objs []EnsureObject) error {
+	var wg sync.WaitGroup
+	errChan := make(chan error)
```

Comment:
```suggestion
	errChan := make(chan error, len(objs))
```

To avoid blocking `errChan <- fmt.Errorf("ensure %+v: %v", obj.Name, err)`